### PR TITLE
Test fix for handshake failure in SSLDriverTests in JDK16

### LINKED ISSUE #67324: SSLDriverTests failures on Java 16
**Build scan**:

https://gradle-enterprise.elastic.co/s/nfsh3tlezlwba

**Repro line**:

```
./gradlew ':x-pack:plugin:security:test' --tests "org.elasticsearch.xpack.security.transport.nio.SSLDriverTests.testHandshakeFailureBecauseProtocolMismatch" -Dtests.seed=E6AA04E3D814FEAE -Dtests.security.manager=true -Dtests.locale=ca -Dtests.timezone=America/St_Barthelemy -Druntime.java=16
```

**Reproduces locally?**:

yes

**Applicable branches**:

I expect all branches running on Java 16

**Failure history**:

https://build-stats.elastic.co/goto/30a7ec831d753f0b753e6c60af3fa5ff

Started failing on January 8th

**Failure excerpt**:

```
   2> java.lang.AssertionError: Unexpected exception message: No appropriate protocol (protocol is disabled or cipher suites are inappropriate)
        at __randomizedtesting.SeedInfo.seed([E6AA04E3D814FEAE:7A3F56720CE43F06]:0)
        at org.junit.Assert.fail(Assert.java:88)
        at org.junit.Assert.assertTrue(Assert.java:41)
        at org.elasticsearch.xpack.security.transport.nio.SSLDriverTests.testHandshakeFailureBecauseProtocolMismatch(SSLDriverTests.java:175)

```

### PR DESCRIPTION
**Pull request by albertzaharovits**
`SSLDriverTests#testHandshakeFailureBecauseProtocolMismatch` intentionally uses mismatching TLS protocols between the client and the server in a test scenario. In this particular case, the mismatch is between "TLSv1.1" and "TLSv1.2".

The problem is that [JDK16](https://jdk.java.net/16/release-notes) does not support "TLSv1.1" .

This fix tests the mismatch between "TLSv1.3" and "TLSv1.2" when running in JDK16 with `-Dtests.fips.enabled=false` and otherwise maintains the original mismatch test between "TLSv1.1" and "TLSv1.2".

Closes https://github.com/elastic/elasticsearch/issues/67324

### CODE REVIEW

File: x-pack/plugin/security/src/test/java/org/elasticsearch/xpack/security/transport/nio/SSLDriverTests.java [hunk 1]
```diff
@@ -22,15 +23,25 @@
...
 import java.util.concurrent.atomic.AtomicInteger;
 import java.util.function.IntFunction;
 
+import static org.hamcrest.Matchers.is;
+
 public class SSLDriverTests extends ESTestCase {
 
+    @Before
+    public void setup() {
+        AccessController.doPrivileged((PrivilegedAction<String>) () -> System.setProperty("jdk.tls.disabledAlgorithms",
```
Comment (jkakavas) [inline x-pack/plugin/security/src/test/java/org/elasticsearch/xpack/security/transport/nio/SSLDriverTests.java:41]: Why do we disable TLSv1.1 in all JDK versions here ?

Comment (albertzaharovits) [reply x-pack/plugin/security/src/test/java/org/elasticsearch/xpack/security/transport/nio/SSLDriverTests.java]: Oops, that's a mistake from a previous approach. I've removed it.

File: x-pack/plugin/security/src/test/java/org/elasticsearch/xpack/security/transport/nio/SSLDriverTests.java [hunk 2]
```diff
@@ -159,20 +164,32 @@ public void testHandshakeFailureBecauseProtocolMismatch() throws Exception {
         SSLEngine clientEngine = sslContext.createSSLEngine();
         SSLEngine serverEngine = sslContext.createSSLEngine();
 
-        String[] serverProtocols = {"TLSv1.2"};
+        final String[] serverProtocols;
+        final String[] clientProtocols;
+        final String expectedMessage;
+        if (inFipsJvm()) {
+            // fips JSSE does not support TLSv1.3 yet
```
Comment (jkakavas) [inline x-pack/plugin/security/src/test/java/org/elasticsearch/xpack/security/transport/nio/SSLDriverTests.java:171]: mental note to revisit this when https://github.com/elastic/elasticsearch/issues/67233 is fixed, as 1.0.11 includes support for TLS1.3

**Review by jkakavas — APPROVED** [review]
LGTM

**Pull request merged**